backend/compact-connect/lambdas/common-python/cc_common/data_model/client.py

```diff
@@ -263,3 +268,121 @@ def create_provider_privileges(
+
+        latest_military_affiliation_record = max(
+            initializing_military_affiliation_records, key=lambda record: record['dateOfUpload']
+        )
+
+        schema = MilitaryAffiliationRecordSchema()
+        with self.config.provider_table.batch_writer() as batch:
+            for record in initializing_military_affiliation_records:
+                if record['dateOfUpload'] == latest_military_affiliation_record['dateOfUpload']:
+                    record['status'] = MilitaryAffiliationStatus.ACTIVE.value
+                else:
+                    record['status'] = MilitaryAffiliationStatus.INACTIVE.value
+
+                serialized_record = schema.dump(record)
+                batch.put_item(Item=serialized_record)
+
+    def create_military_affiliation(
+        self,
+        compact: str,
+        provider_id: str,
+        affiliation_type: MilitaryAffiliationType,
+        file_names: list[str],
+        document_keys: list[str],
+    ):
+        """
+        Create a new military affiliation record for a provider in the database.
+
+        If there are any previous active military affiliations for this provider, they will be set to inactive.
+
+        :param compact: The compact name
+        :param provider_id: The provider id
+        :param affiliation_type: The type of military affiliation
+        :param file_names: The list of file names for the documents
+        :param document_keys: The list of s3 document keys for the documents
+        :return: The created military affiliation record
+        """
+
+        latest_military_affiliation_record = {
+            'type': 'militaryAffiliation',
+            'affiliationType': affiliation_type.value,
+            'fileNames': file_names,
+            'compact': compact,
+            'providerId': provider_id,
+            # we set this to initializing until the client uploads the document, which
+            # will trigger another lambda to update the status to active
+            'status': MilitaryAffiliationStatus.INITIALIZING.value,
+            'documentKeys': document_keys,
+            'dateOfUpload': datetime.now(tz=UTC),
+        }
+
+        schema = MilitaryAffiliationRecordSchema()
+        latest_military_affiliation_record_serialized = schema.dump(latest_military_affiliation_record)
+
+        # we need to check if any other active military affiliations exist for this provider
+        # if they do, we need to set them to inactive
+        active_military_affiliation_records = self._get_active_military_affiliation_records(compact, provider_id)
+
+        with self.config.provider_table.batch_writer() as batch:
+            batch.put_item(Item=latest_military_affiliation_record_serialized)
+
+            for record in active_military_affiliation_records:
+                record['status'] = MilitaryAffiliationStatus.INACTIVE.value
+                serialized_record = schema.dump(record)
+                batch.put_item(Item=serialized_record)
```

Comment:
Yeah that could work, it just means we would be making two API batch calls to Dynamo, rather than a single batch call. So slightly less performant, but not by much.

Reply:
I could go either way, let me know if you have a strong preference and I'll update as needed.

Reply:
I vote the two API call approach - perhaps slightly less performant, but more readable imo. Perhaps add a brief comment above the second call like `// These could be consolidated into a single batch call if performance becomes an issue`